### PR TITLE
Clean up monster.cpp

### PR DESCRIPTION
**Pull request by kphoenix137**

### CODE REVIEW

**Comment by kphoenix137** [comment]
Do we need a new timedemo or did I mess up bad

File: Source/monster.cpp [hunk 1]
```diff
@@ -55,9 +55,8 @@ size_t LevelMonsterTypeCount;
 Monster Monsters[MaxMonsters];
 int ActiveMonsters[MaxMonsters];
 size_t ActiveMonsterCount;
-// BUGFIX: replace MonsterKillCounts[MaxMonsters] with MonsterKillCounts[NUM_MTYPES].
 /** Tracks the total number of monsters killed per monster_id. */
-int MonsterKillCounts[MaxMonsters];
+int MonsterKillCounts[NUM_MTYPES];
```
Comment (StephenCWills) [inline Source/monster.cpp:59]: You need to make sure the save file skips/pads the difference.
https://github.com/diasurgical/devilutionX/blob/c534e0e93c07d6d6d85151eeccac1d5699b988da/Source/loadsave.cpp#L2096-L2097
https://github.com/diasurgical/devilutionX/blob/c534e0e93c07d6d6d85151eeccac1d5699b988da/Source/loadsave.cpp#L2350-L2351

**Comment by AJenbo** [comment]
> Do we need a new timedemo or did I mess up bad

did you intend to change game play?

**Comment by kphoenix137** [comment]
> > Do we need a new timedemo or did I mess up bad
> 
> did you intend to change game play?

No, just applying bugfix. I'll do Stephen's suggestion and pad bytes

File: Source/monster.cpp [hunk 2]
```diff
@@ -1850,33 +1850,34 @@ void AiAvoidance(Monster &monster)
 MissileID GetMissileType(MonsterAIID ai)
 {
 	switch (ai) {
-	case MonsterAIID::GoatMelee:
+	using enum MonsterAIID;
```
Comment (AJenbo) [inline Source/monster.cpp]: This is C++20 only, but we need to support some targets that barely C++17

Comment (AJenbo) [reply Source/monster.cpp]: ... but looks like it did work, so feel fee to keep it like this.

Comment (kphoenix137) [reply Source/monster.cpp]: Didn't compile

**Comment by AJenbo** [comment]
do you know how to run the demo locally with visuals and debugging?

**Comment by kphoenix137** [comment]
> do you know how to run the demo locally with visuals and debugging?
> 

Naur

File: Source/loadsave.cpp [hunk 3]
```diff
@@ -2096,6 +2096,10 @@ void LoadGame(bool firstflag)
 	for (int &monstkill : MonsterKillCounts)
 		monstkill = file.NextBE<int32_t>();
 
+	// add padding for vanilla save compatibility (Related to bugfix where MonsterKillCounts[MaxMonsters] is was changed to MonsterKillCounts[NUM_MTYPES]
+	int padZero = 0;
+	for (int i = 0; i < MaxMonsters - NUM_MTYPES; i++)
+		padZero = file.NextBE<int32_t>();
```
Comment (StephenCWills) [inline Source/loadsave.cpp:2102]:
```suggestion
	// skip ahead for vanilla save compatibility (Related to bugfix where MonsterKillCounts[MaxMonsters] was changed to MonsterKillCounts[NUM_MTYPES]
	file.Skip(4 * (MaxMonsters - NUM_MTYPES));
```

File: Source/loadsave.cpp [hunk 4]
```diff
@@ -2349,6 +2353,9 @@ void SaveGameData(SaveWriter &saveWriter)
 		SavePortal(&file, i);
 	for (int monstkill : MonsterKillCounts)
 		file.WriteBE<int32_t>(monstkill);
+	// add padding for vanilla save compatibility (Related to bugfix where MonsterKillCounts[MaxMonsters] is was changed to MonsterKillCounts[NUM_MTYPES]
+	for (int i = 0; i < MaxMonsters - NUM_MTYPES; i++)
+		file.WriteBE<int32_t>(0);
```
Comment (StephenCWills) [inline Source/loadsave.cpp:2358]:
```suggestion
	// add padding for vanilla save compatibility (Related to bugfix where MonsterKillCounts[MaxMonsters] was changed to MonsterKillCounts[NUM_MTYPES]
	file.Skip(4 * (MaxMonsters - NUM_MTYPES));
```

**Review by StephenCWills — CHANGES_REQUESTED** [review]
I think you can just use `file.Skip()` in both places to avoid the loops.

File: Source/loadsave.cpp [hunk 5]
```diff
@@ -2096,6 +2096,10 @@ void LoadGame(bool firstflag)
 	for (int &monstkill : MonsterKillCounts)
 		monstkill = file.NextBE<int32_t>();
 
+	// add padding for vanilla save compatibility (Related to bugfix where MonsterKillCounts[MaxMonsters] is was changed to MonsterKillCounts[NUM_MTYPES]
```
Comment (qndel) [inline Source/loadsave.cpp:2099]: is or was

Comment (kphoenix137) [reply Source/loadsave.cpp:2099]: Were

Comment (AJenbo) [reply Source/loadsave.cpp:2099]: " is was changed"

Comment (AJenbo) [reply Source/loadsave.cpp:2099]:
```suggestion
	// Add padding for vanilla save compatibility (Related to bugfix where MonsterKillCounts[MaxMonsters] was changed to MonsterKillCounts[NUM_MTYPES]

```

Comment (kphoenix137) [reply Source/loadsave.cpp:2099]: > "MonsterKillCounts[MaxMonsters] is was changed"

🫠

**Review by AJenbo — DISMISSED** [review]
Nice work, sorry for the long delay in reviewing it.

**Pull request merged**